src/backend/joanie/core/api/admin.py

```diff
@@ -492,12 +492,11 @@ def generate_certificate(self, request, pk=None):  # pylint:disable=unused-argum
         Generate the certificate for an order when it is eligible.
         """
         order = self.get_object()
-
-        certificate, created = order.get_or_generate_certificate()
-
-        if not certificate:
+        try:
+            certificate, created = order.get_or_generate_certificate()
+        except ValidationError as error:
             return JsonResponse(
-                {"details": f"Cannot issue certificate for order {order.id}"},
```

Comment:
IMO the `details` key should be kept. And it should contains a dict with the `code` and the `message`

Reply:
I might not understand what you suggest for this point.

Should I return a dictionary such as : 
`JsonResponse{ "details" : {"product" : "Certificate cannot be generated"} ` ?

Why should i keep the `details` key ?

Reply:
I thought DRF was doing this but I don't find in the document if it's the case or not.
You can remove the `details` key and forget my comment.

Reply:
I think we should keep details to match ErrorResponseSerializer, or add a custom serializer.

Reply:
In order to simplify this implementation, the error message will be a string only.
The key will remain "details" to match openapi schema.